README.md

```diff
@@ -51,6 +78,33 @@ Quick Start
 Trouble with these steps?  Read `TROUBLESHOOTING.md`.  More detailed build
 guidance is available below.
 
+**Submodule**
+
+If your project build tool is CMake, you can add libgit2 as a dependency on your project using `Git submodule`
+
+```
+cd /path/your/repo
+git submodule add https://github.com/libgit2/libgit2.git vendor/libgit2
+cd vendor/libgit2
+git checkout LATEST_RELEASE
+cd ../..
+```
+Modify the contents of the CMakeLists.txt file:
+
+```cmake
+# suggest use set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+include_directories(
+    ./include
+    ./vendor/libgit2/include
+)
+add_subdirectory(vendor/libgit2)
```

Comment:
Even though these instructions are targeted at other projects, I'd like to keep our own style here:

```
# Add the libgit2 library
ADD_SUBDIRECTORY(vendor/libgit2)
# Add libgit2 headers to your executable's include directory
TARGET_INCLUDE_DIRECTORIES(${YOUR_TARGET} ${LIBGIT2_INCLUDES}
# Link your executable against libgit2
TARGET_LINK_LIBRARIES(${YOUR_TARGET} git2)
```